### ISSUE TITLE
1.5 Hours Course

### ISSUE THREAD

**Issue by anish-kum** [anish-kum/PPP-Spark-2025-2026#11]
90 min PPP

**Comment by anish-kum** [comment]
<img width="334" height="438" alt="Image" src="https://github.com/user-attachments/assets/4da630cd-af2d-41da-8152-f30a053ee272" />

<img width="326" height="348" alt="Image" src="https://github.com/user-attachments/assets/fb4a22e1-3cb1-4629-b863-dc26d907ffa6" />

What I was learning:

- Model Evaluation: Discusses the importance of evaluating model performance using metrics like accuracy, precision, recall, and F1 score.

- Train-Validation Split: Introduces the concept of splitting the dataset into training and validation sets to assess model performance on unseen data.

- Overfitting and Underfitting: Explains the concepts of overfitting (model learns noise) and underfitting (model is too simple), emphasizing the need for balance.

- Regularization Techniques: Covers methods like L1 and L2 regularization to prevent overfitting by adding penalties to the loss function.

- Hyperparameter Tuning: Discusses the process of optimizing hyperparameters (learning rate, batch size) to improve model performance.

- Cross-Validation: Introduces k-fold cross-validation as a technique to ensure robust evaluation by training multiple models on different subsets of the data.

- Final Model Selection: Emphasizes the importance of selecting the best model based on evaluation metrics and validation performance.

Questions the content didn't cover that I'd like an answer to:

- How do you handle imbalanced datasets when evaluating models?

- When should you use cross-validation vs. a separate test set for final model evaluation?

Note: This was a lot this week, and I spent more than 90 minutes on this. There's a lot of syntax that I don't have a full understanding of yet, but as I code, I'll be able to naturally intuit how to approach a task. I don't want to just memorize different functions, etc, so getting a full understanding will take a lot more time.

**Comment by drewmca-pkwy** [comment]
@anish-kum I appreciate the reflection here, Anish. When you use more than your 90 minutes, be sure add that in the "Estimate" field so we can get a sense of how much effort you're putting toward this. I appreciate the questions you're generating, and I look forward to what your PPP Guide will have to say about them!
